KD_Lib/KD/text/Bert2Lstm/bert2lstm.py

```diff
@@ -0,0 +1,217 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import random
+
+from transformers import BertForSequenceClassification, AdamW, BertTokenizer
+
+from KD_Lib.common import BaseClass
+from KD_Lib.KD.text.utils.bert import train_bert, evaluate_bert, get_bert_dataloader
+from KD_Lib.KD.text.utils.lstm import train_lstm, evaluate_lstm, distill_to_lstm
+
+import numpy as np
+import matplotlib.pyplot as plt
+from copy import deepcopy
+
+
+class Bert2LSTM(BaseClass):
+    """
+    Implementation of Knowledge distillation from the paper "Distilling Task-Specific
+    Knowledge from BERT into Simple Neural Networks" https://arxiv.org/pdf/1903.12136.pdf
+
+    :param student_model (torch.nn.Module): Student model
+    :param distill_train_loader (torch.utils.data.DataLoader): Student Training Dataloader for distillation
+    :param distill_val_loader (torch.utils.data.DataLoader): Student Testing/validation Dataloader
+    :param optimizer_student (torch.optim.*): Optimizer used for training student
+    :param train_df (pandas.DataFrame): Dataframe for training the teacher model
+    :param val_df (pandas.DataFrame): Dataframe for validating the teacher model
+    :param optimizer_teacher (torch.optim.*): Optimizer used for training teacher. If None, AdamW is used.
+    :param loss_fn (torch.nn.module): Loss function
+    :param temp (float): Temperature parameter for distillation
+    :param distil_weight (float): Weight paramter for distillation loss
+    :param device (str): Device used for training; 'cpu' for cpu and 'cuda' for gpu
+    :param log (bool): True if logging required
+    :param logdir (str): Directory for storing logs
+    """
+
+    def __init__(
+        self,
+        student_model,
+        distill_train_loader,
+        distill_val_loader,
+        train_df,
+        val_df,
+        num_classes=2,
+        seed=42,
+        loss_fn=nn.MSELoss(),
+        temp=20.0,
+        distil_weight=0.5,
+        device="cpu",
+        log=False,
+        logdir="./Experiments",
+    ):
+
+        self.set_seed(42)
+
+        self.train_df, self.val_df = train_df, val_df
+
+        teacher_model = BertForSequenceClassification.from_pretrained(
+            "bert-base-uncased",
+            num_labels=num_classes,
+            output_attentions=False,
+            output_hidden_states=False,
+        )
+
+        optimizer_teacher = AdamW(teacher_model.parameters(), lr=2e-5, eps=1e-8)
+
+        self.bert_tokenizer = BertTokenizer.from_pretrained(
+            "bert-base-uncased", do_lower_case=True
+        )
+
+        super(Bert2LSTM, self).__init__(
+            teacher_model,
+            student_model,
+            distill_train_loader,
+            distill_val_loader,
+            optimizer_teacher,
+            None,
+            loss_fn,
+            temp,
+            distil_weight,
+            device,
+            log,
+            logdir,
+        )
```

Comment:
Please shift this to the beginning of the __init__ method this is the proper flow of calling the base class

Reply:
Thank you for the detailed review. 
1. I am not very familiar with tensorboard and hence did not include it here. I would request someone else to possibly take this up or maybe I can do it later.
2. That seems like a nice approach. Are you suggesting that we have a class for the model which handles its training/distillation, etc.? 
3. I will change that. Thank you for pointing that out.